pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -17,275 +17,319 @@ describe('LiquidityBootstrappingPool', function () {
-          now = await currentTimestamp();
-          startTime = now.add(START_DELAY);
-          endTime = startTime.add(UPDATE_DURATION);
-
-          await pool.updateWeightsGradually(owner, startTime, endTime, endWeights);
-        });
-
-        it('stores the params', async () => {
-          const updateParams = await pool.getGradualWeightUpdateParams();
-
-          expect(updateParams.startTime).to.equalWithError(startTime, 0.001);
-          expect(updateParams.endTime).to.equalWithError(endTime, 0.001);
-          expect(updateParams.endWeights).to.equalWithError(endWeights, 0.001);
-        });
-
-        it('gets start weights if called before the start time', async () => {
-          const normalizedWeights = await pool.getNormalizedWeights();
-
-          // Need to decrease precision
-          expect(normalizedWeights).to.equalWithError(pool.normalizedWeights, 0.0001);
+      describe('public actions', () => {
+        sharedBeforeEach('set sender to other', async () => {
+          sender = other;
         });
 
-        it('gets end weights if called after the end time', async () => {
-          await advanceTime(endTime.add(MINUTE));
-          const normalizedWeights = await pool.getNormalizedWeights();
-
-          // Need to decrease precision
-          expect(normalizedWeights).to.equalWithError(endWeights, 0.0001);
-        });
-
-        it('gets intermediate weights if called halfway through', async () => {
-          await advanceTime(START_DELAY + UPDATE_DURATION / 2);
-          const normalizedWeights = await pool.getNormalizedWeights();
-
-          // Need to decrease precision
-          expect(normalizedWeights).to.equalWithError(halfWeights, 0.001);
+        it('anyone can get the swap state', async () => {
+          await expect(pool.getSwapEnabled(sender)).to.not.be.reverted;
         });
-
-        it('gets intermediate weights if called 20% through', async () => {
-          await advanceTime(START_DELAY + UPDATE_DURATION * 0.2);
-          const normalizedWeights = await pool.getNormalizedWeights();
-
-          // Need to decrease precision
-          expect(normalizedWeights).to.equalWithError(oneFifthWeights, 0.001);
+  
+        it('anyone can swap', async () => {
+          await pool.init({ from: owner, initialBalances });
+  
+          await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(0.1) })).to.not.be.reverted;
         });
 
-        it('gets intermediate weights if called 80% through', async () => {
-          await advanceTime(START_DELAY + UPDATE_DURATION * 0.8);
-          const normalizedWeights = await pool.getNormalizedWeights();
-
-          // Need to decrease precision
-          expect(normalizedWeights).to.equalWithError(fourFifthWeights, 0.001);
-        });
+        it('anyone can query the weight update params', async () => {
+          await expect(pool.getGradualWeightUpdateParams(sender)).to.not.be.reverted;
+        })
```

Comment:
I'd remove the tests for the `view` functions, it is expected that they don't depend on the caller.